# Here is a list of DMS studies to add to the repository:  

## Unsorted

[forsyth2013deep](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3906306/)  
*Deep mutational scanning of an antibody against epidermal growth factor receptor using mammalian cell display and massively parallel pyrosequencing*  
antibody binding domain

[adams2016measuring](https://elifesciences.org/content/5/e23156)  
*Measuring the sequence-affinity landscape of antibodies with massively parallel titration curves*   
scFv antibody

[koenig2017mutational](http://www.pnas.org/content/114/4/E486.abstract)    
*Mutational landscape of antibody variable domainsreveals a switch modulating the interdomainconformational dynamics and antigen binding*  
anti-VEGF antibody G6.31   

[wrenbeck2017single](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5467163/)  
*Single-mutation fitness landscapes for an enzyme on multiple substrates reveal specificity is globally encoded*  
amiE

[bandaru2017deconstruction](https://elifesciences.org/articles/27810)  
*Deconstruction of the Ras switching cycle through saturation mutagenesis*  
ras

## Easy to find alignments

## Easy to get preferences

[olson2014comprehensive](http://ac.els-cdn.com/S0960982214012688/1-s2.0-S0960982214012688-main.pdf?_tid=efadb44e-14bc-11e7-8c18-00000aacb35d&acdnat=1490818916_3fa220dec513e9c0711b9f718bd0ed0f)  
*A comprehensive biophysical description of pairwise epistasis throughout an entire protein domain*  
GB1  

[tripathi201molecular](https://academic.oup.com/mbe/article/33/11/2960/2272479/Molecular-Determinants-of-Mutant-Phenotypes#45198683)    
*Molecular Determinants of Mutant Phenotypes, Inferred from Saturation Mutagenesis Data*  
CcdB  
have link to supp materials but the link is not working

## Easy to find alignment *and* easy to get preferences

[haddox2016experimental](http://journals.plos.org/plospathogens/article/file?id=10.1371/journal.ppat.1006114&type=printable)  
*Experimental estimation of the effects of all amino-acid mutations to HIV’s envelope protein on viral replication in cell culture*  
env

[doud2015site](https://academic.oup.com/mbe/article/32/11/2944/982113/Site-Specific-Amino-Acid-Preferences-Are-Mostly)  
*Site-specific amino acid preferences are mostly conserved in two closely related protein homologs*  
NP

[brenan2016phenotypic](http://ac.els-cdn.com/S2211124716313171/1-s2.0-S2211124716313171-main.pdf?_tid=63164508-14be-11e7-906d-00000aab0f6b&acdnat=1490819540_78812c05444e95cbd7db52bfb4b6fdac)    
*Phenotypic characterization of a comprehensive set of MAPK1/ERK2 missense mutants*  
MAPK1/ERK2  
[orth](http://uswest.ensembl.org/Homo_sapiens/Gene/Compara_Ortholog?db=core;g=ENSG00000100030;r=22:21754500-21867680)

[kitzman2015massively](http://www.nature.com/nmeth/journal/v12/n3/pdf/nmeth.3223.pdf)  
*Massively parallel single-amino-acid mutagenesis*  
Gal4  
*sequences from bloom2017identification*

## Raw data hard to find  
[mclaughlin2012spatial](http://www.nature.com/nature/journal/v491/n7422/abs/nature11500.html)   
*The spatial architecture of protein function and
adaptation*  
PDZ domain   
data in a `MATLAB` workspace    

[melnikov2014comprehensive](https://academic.oup.com/nar/article/42/14/e112/1266940/Comprehensive-mutational-scanning-of-a-kinase-in)   
*Comprehensive mutational scanning of a kinase in vivo reveals substrate-dependent fitness landscapes*  
APH(3′)II    
supp page wasn't loading

[majithia2016prospective](http://www.nature.com/ng/journal/v48/n12/abs/ng.3700.html)    
*Prospective functional classification of all possible missense variants in PPARG*  
PPARG  
hard to find the raw data  
[orth](http://uswest.ensembl.org/Homo_sapiens/Gene/Compara_Ortholog?db=core;g=ENSG00000132170;r=3:12287368-12434356)

[klesmith2017tradeoffs](http://www.pnas.org/content/114/9/2265.full.pdf)  
*Trade-offs between enzyme fitness and solubility illuminated by deep mutational scanning*  
beta and LGK   
couldn't find link to raw data

[kelsic2016rna](http://www.sciencedirect.com.offcampus.lib.washington.edu/science/article/pii/S2405471216303684)  
*RNA Structural Determinants of Optimal Codons Revealed by MAGE-Seq*  
infA  
data wouldn't download
