Here is a list of DMS studies to add to the repository:  

[mclaughlin2012spatial](http://www.nature.com/nature/journal/v491/n7422/pdf/nature11500.pdf)   
*The spatial architecture of protein function and
adaptation*  
PDZ domain

[roscoe2014Analyses](http://ac.els-cdn.com/S0022283613000636/1-s2.0-S0022283613000636-main.pdf?_tid=4c8ee90e-14bc-11e7-9642-00000aacb362&acdnat=1490818643_3730c928506e536d8375dee44c8e426f)  
*Analyses of the Effects of All Ubiquitin Point Mutants on
Yeast Growth Rate*  
ubiquitin   

[olson2014comprehensive](http://ac.els-cdn.com/S0960982214012688/1-s2.0-S0960982214012688-main.pdf?_tid=efadb44e-14bc-11e7-8c18-00000aacb35d&acdnat=1490818916_3fa220dec513e9c0711b9f718bd0ed0f)  
*A comprehensive biophysical description of pairwise epistasis throughout an entire protein domain*  
GB1  

[melnikov2014comprehensive](https://oup.silverchair-cdn.com/oup/backfile/Content_public/Journal/nar/42/14/10.1093/nar/gku511/3/gku511.pdf?Expires=1491239445&Signature=SOPm6QvYXYSXhEyEYeBgeFeOxoIgmiL-k7e3Uvk6qTSvfbDAQYjHcqNg39Qp-GJ7Ea7ads2EBBzs2OTEAIFHU8c5QgL2yeoI63SIJw03F95SubrgKOJGjP7E6s4AENW7MrRgwIVAS-xb0SorfDznDyLokv-4Tivq4FSC6OCYKjRHq4ZY2N9~hO3UGx1PNgcXJzQfzFb407NDhasyHZXM~PipXOWkacOHwzj-1ToZ9tb6sQMUkuSxiKH5R0lYRcEHkJ4Abmzcl6q6sj~Z1Rj0xHYcTKryfsFlTsQAJaURigF3ogOW2WF31ny38BJNYpxRpvgHFEJN9~cgkTyvkB2jeA__&Key-Pair-Id=APKAIUCZBIA4LVPAVW3Q)   
*Comprehensive mutational scanning of a kinase in vivo reveals substrate-dependent fitness landscapes*  
APH(3′)II  

[doud2015site](https://academic.oup.com/mbe/article/32/11/2944/982113/Site-Specific-Amino-Acid-Preferences-Are-Mostly)  
*Site-specific amino acid preferences are mostly conserved in two closely related protein homologs*  
NP

[kitzman2015massively](http://www.nature.com/nmeth/journal/v12/n3/pdf/nmeth.3223.pdf)  
*Massively parallel single-amino-acid mutagenesis*  
Gal4, p53

[mishra2016systematic](http://ac.els-cdn.com/S2211124716303175/1-s2.0-S2211124716303175-main.pdf?_tid=ceb214be-14bd-11e7-8d9f-00000aacb35e&acdnat=1490819290_99c62c0c6d511fad788c96ef51056d76)    
*Systematic mutant analyses elucidate general and client-specific aspects of Hsp90 function*  
Hsp90


[mavor2016determination](https://elifesciences.org/content/5/e15802)  
*Determination of ubiquitin fitness landscapes under different chemical stresses in a classroom setting*    
ubiquitin

[haddox2016experimental](http://journals.plos.org/plospathogens/article/file?id=10.1371/journal.ppat.1006114&type=printable)  
*Experimental estimation of the effects of all amino-acid mutations to HIV’s envelope protein on viral replication in cell culture*  
env

[majithia2016prospective](http://www.nature.com/ng/journal/v48/n12/pdf/ng.3700.pdf)    
*Prospective functional classification of all possible missense variants in PPARG*  
PPARG

[brenan2016phenotypic](http://ac.els-cdn.com/S2211124716313171/1-s2.0-S2211124716313171-main.pdf?_tid=63164508-14be-11e7-906d-00000aab0f6b&acdnat=1490819540_78812c05444e95cbd7db52bfb4b6fdac)    
*Phenotypic characterization of a comprehensive set of MAPK1/ERK2 missense mutants*  
MAPK1/ERK2

[klesmith2017tradeoffs](http://www.pnas.org/content/114/9/2265.full.pdf)  
*Trade-offs between enzyme fitness and solubility illuminated by deep mutational scanning*  
beta and LGK

[kelsic2016rna](http://www.cell.com/cell-systems/pdf/S2405-4712(16)30368-4.pdf)  
*RNA Structural Determinants of Optimal Codons Revealed by MAGE-Seq*  
infA

[koenig2017mutational](http://www.pnas.org/content/114/4/E486.full.pdf)    
*Mutational landscape of antibody variable domainsreveals a switch modulating the interdomainconformational dynamics and antigen binding*  
anti-VEGF antibody G6.31   

[tripathi201molecular](https://oup.silverchair-cdn.com/oup/backfile/Content_public/Journal/mbe/33/11/10.1093_molbev_msw182/2/msw182.pdf?Expires=1491347672&Signature=XJwz9Ho1R8-ssIBXp4kgfaU1AwlgiNu7nuYmriEGmIkS2YRgbfcyOBMe0NkbyvZ-TP8p24xgzCX6k9RbtSO0DQLjEsi17IGOqvBgifBTbcc6ubrByOf4DmP-mjfQhfjAB183ahU0oNM3Q-6nP7qjQeaZOMTi8w2yDwA1ySLYbuEdJBVunOY7-F3Jg6tccayYvePtHK3H-LB5YYa1ZmG3nTnNwtdyxZGIpB9256tiqXvfOvzNXvm~JMBfkOw8nJiMWfd6v7f6qysLSMSP4bZ2nfiimUaGmBbNZCCk0danQbpuHEe6NRLYZxytE3ltUWajOTuOKBLf5xa36xpCGuXnqg__&Key-Pair-Id=APKAIUCZBIA4LVPAVW3Q)    
*Molecular Determinants of Mutant Phenotypes, Inferred from Saturation Mutagenesis Data*  
CcdB  

[kowalsky2016determination](http://onlinelibrary.wiley.com.offcampus.lib.washington.edu/store/10.1002/prot.25175/asset/prot25175.pdf;jsessionid=115A7C62D6F5E84C80D8310770D63E8B.f03t01?v=1&t=j18fbo4l&s=50a94a32127bd7ed2951cdceba4b75502a7a604a)  
*Determination of binding affinity upon
mutation for type I dockerin–cohesin
complexes from Clostridium thermocellum and
Clostridium cellulolyticum using deep sequencing*  
type I cohesin
